Htsget chart

aaperis opened the pull request:
This PR adds the Helm Chart for `htsget-rs` and closes https://github.com/NBISweden/LocalEGA-SE-Deployment/issues/755.

The chart is not restricted to GDI usecases, but covers all possible configurations mentioned in the [htsget-rs repo](https://github.com/umccr/htsget-rs/tree/main/htsget-config). A README file with all variables explained is included. 
Given that many configuration features of htsget-rs rely heavily on a toml file with `ini` structure, the chart is set up around this dependency and leverage of environment variables for configuring the app is kept to bare minimum.

I have also included tests and integration tests (where possible) for deploying htsget-rs in different storage backend scenarions, see at the end of  README for details.

Before testing manually, take a look at the integration tests where many cases are covered. If you'd still like to test manually you can use minikube or k3d. Details for k3d can be found in the scripts under `.github/integration/scripts` or in the repo's [wiki](https://github.com/NBISweden/htsget-rs-helm/wiki).

aaperis committed:
create boilerplate

aaperis committed:
add htsget specifics

aaperis committed:
set default image tag in chart

aaperis committed:
update configmap with latest config

aaperis committed:
update readme

pahatz commented on README.md:
```diff
@@ -1,2 +1,79 @@
+
+### Installing the htsget-rs helm chart
+
+Run `helm install charts/htsget-rs` to install the system with the `htsget-rs` prefix.
```
```suggestion
Run `helm install htsget-rs charts/htsget-rs` to install the  `htsget-rs` chart.
```

pahatz commented on README.md:
```diff
@@ -1,2 +1,79 @@
+
+### Testing with sda-download
+
+Start by checking out [this branch}(https://github.com/GenomicDataInfrastructure/starter-kit-storage-and-interfaces/tree/feat/reencrypt-header) of the GDI starter-kit-storage&interfaces repository. This branch contains the re-encryption of file headers in the sda-download service. Then run:
```
```suggestion
Start by checking out [this branch}(https://github.com/GenomicDataInfrastructure/starter-kit-storage-and-interfaces/tree/featute/test-htsget-reencrypt) of the GDI starter-kit-storage&interfaces repository. This branch contains the re-encryption of file headers in the sda-download service.
Follow the instructions in the README of that repository to set-up all necessary configuration files. Then run:
```

pahatz commented on README.md:
````diff
@@ -1,2 +1,79 @@
+```sh
+docker compose -f docker-compose-demo.yml up
+```
+and wait until the `data_loader` comletes with `exit 0`.
````
```suggestion
and wait until the `data_loader` completes with `exit 0`.
```

pahatz commented on README.md:
````diff
@@ -1,2 +1,79 @@
+```
+and wait until the `data_loader` comletes with `exit 0`.
+
+Then paste the following into charts/htsget-rs/configmap.yaml:
````
```suggestion
Then paste the following into charts/htsget-rs/templates/configmap.yaml:
```

pahatz replied:
already there?

aaperis committed:
Apply suggestions from code review

Co-authored-by: Panos Chatzopoulos <panos.chatzopoulos@nbis.se>

aaperis committed:
make port name app specific

aaperis committed:
add tls logic

- ingress tls
- certificate for mTLS
- update values file

aaperis committed:
configure htsget from values file

- add option to use toml config in values file

aaperis committed:
enable htsget tls configuration

aaperis committed:
add c4gh configuration

aaperis committed:
move values configuration to deployment template

- move readiness and liveness probes
- move volume directives

aaperis committed:
separate client and server tls configurations

aaperis committed:
add cert-manager for tests

aaperis committed:
update readme

aaperis committed:
fix config for enabling  c4gh encryption

aaperis committed:
enable ingress tls from values file

aaperis committed:
update default image tag

aaperis committed:
configure logging with env variables

aaperis committed:
read server certs on tls enabled

aaperis committed:
add role label in template

aaperis committed:
add deployment tests

aaperis committed:
add chart publishing

aaperis committed:
cleanup tests

jbygdell commented on charts/htsget-rs/Chart.yaml:
```diff
@@ -0,0 +1,13 @@
+apiVersion: v2
+name: htsget-rs
+description: A Helm chart for Kubernetes
```
Helm charts by default are used with kubernetes, this description should be about the application.

jbygdell commented on charts/htsget-rs/templates/certificate.yaml:
```diff
@@ -0,0 +1,43 @@
+    size: 256
+  usages:
+    - server auth
+    - client auth
```
```suggestion
```

HTSGET have no use for a client certificate since it has no mTLS partner

jbygdell commented on charts/htsget-rs/templates/certificate.yaml:
```diff
@@ -0,0 +1,43 @@
+  isCA: false
+  privateKey:
+    algorithm: ECDSA
+    size: 256
```
```suggestion
    algorithm: Ed25519
```

Use Ed25519 cerfificates or at least 384 byte keys if you use ECDSA

jbygdell commented on .github/integration/scripts/values.yaml:
```diff
@@ -0,0 +1,164 @@
+  # the secrets containing them need to be created manually.
+   - secretName: htsget.local-tls
+     hosts:
+       - htsget.local
```
```suggestion
  tls: []
  # If the certificates are generated by external providers
  # the secrets containing them need to be created manually.
  # - secretName: htsget.local-tls
  #   hosts:
  #     - htsget.local
```

jbygdell commented on charts/htsget-rs/templates/ingress.yaml:
```diff
@@ -0,0 +1,70 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "htsget-rs.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
```
```suggestion
apiVersion: networking.k8s.io/v1
```

Remember that you are targeting current stable k8s versions >= 1.26

aaperis replied:
In order not to exclude entirely the possibility of deploying the chart in older k8s clusters, we could release a first chart version keeping the  back-compatibility as is now and then move to `kubeVersion: "^1.26.0-0"` with a subsequent release.

aaperis replied:
Made changes to support >=1.19.0 k8s for now.

jbygdell commented on charts/htsget-rs/templates/ingress.yaml:
```diff
@@ -0,0 +1,70 @@
+              {{- else }}
+              serviceName: {{ $fullName }}
+              servicePort: {{ $svcPort }}
+              {{- end }}
```
```suggestion
```

jbygdell commented on charts/htsget-rs/templates/ingress.yaml:
```diff
@@ -0,0 +1,70 @@
+            pathType: {{ .pathType }}
+            {{- end }}
+            backend:
+              {{- if semverCompare ">=1.19-0" $.Capabilities.KubeVersion.GitVersion }}
```
```suggestion
```

jbygdell commented on charts/htsget-rs/templates/ingress.yaml:
```diff
@@ -0,0 +1,70 @@
+        paths:
+          {{- range .paths }}
+          - path: {{ .path }}
+            {{- if and .pathType (semverCompare ">=1.18-0" $.Capabilities.KubeVersion.GitVersion) }}
```
```suggestion
```

jbygdell commented on charts/htsget-rs/templates/ingress.yaml:
```diff
@@ -0,0 +1,70 @@
+          - path: {{ .path }}
+            {{- if and .pathType (semverCompare ">=1.18-0" $.Capabilities.KubeVersion.GitVersion) }}
+            pathType: {{ .pathType }}
+            {{- end }}
```
```suggestion
```

jbygdell commented on charts/htsget-rs/templates/serviceaccount.yaml:
```diff
@@ -0,0 +1,13 @@
+{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "htsget-rs.serviceAccountName" . }}
+  labels:
+    {{- include "htsget-rs.labels" . | nindent 4 }}
+  {{- with .Values.serviceAccount.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+automountServiceAccountToken: {{ .Values.serviceAccount.automount }}
+{{- end }}
```
```suggestion
```

Service accounts are not needed any longer unless the application needs to talk to the k8s API or you require some extra features that is not part of the Pod Security Standard.

aaperis replied:
I'll leave this here for now since we will support older k8s versions for the first chart version at least.

jbygdell commented on charts/htsget-rs/templates/ingress.yaml:
```diff
@@ -0,0 +1,70 @@
+    {{ toYaml .Values.ingress.annotations | indent 4 }}
+    {{- end }}
+spec:
+  {{- if and .Values.ingress.className (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion) }}
```
```suggestion
```

jbygdell commented on charts/htsget-rs/templates/ingress.yaml:
```diff
@@ -0,0 +1,70 @@
+spec:
+  {{- if and .Values.ingress.className (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion) }}
+  ingressClassName: {{ .Values.ingress.className }}
+  {{- end }}
```
```suggestion
```

jbygdell commented on charts/htsget-rs/templates/hpa.yaml:
```diff
@@ -0,0 +1,32 @@
+  metrics:
+    {{- if .Values.autoscaling.targetCPUUtilizationPercentage }}
+    - type: Resource
+      resource:
+        name: cpu
+        target:
+          type: Utilization
+          averageUtilization: {{ .Values.autoscaling.targetCPUUtilizationPercentage }}
```
How do you know this is the limiting factor?

aaperis replied:
I don't, there is an `if` clause on line 16 above. In the `values.yaml` around line 81 one can control how cpu,  ram or both will be utilized. Maybe `hpa` is a bit too much to include for the current app but it is quite standard to have it as far as I've seen.

jbygdell commented on charts/htsget-rs/values.yaml:
```diff
@@ -0,0 +1,164 @@
+nameOverride: ""     #
+fullnameOverride: "" #
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Automatically mount a ServiceAccount's API credentials?
+  automount: true
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
```
```suggestion
```

Not needed any longer

aaperis replied:
I will leave it there for now for legacy reasons but set default to `create: false`.

jbygdell commented on charts/htsget-rs/values.yaml:
```diff
@@ -0,0 +1,164 @@
+  sendEncryptedToClient: false # set to true if c4gh encryption is used
+  # The url that will be used for the client's url
+  # responseUrl: "http://host.minikube.internal:8443/s3-enrcypted/"
+  responseUrl: "http://host.minikube.internal:8443/s3/"
```
```suggestion
  responseUrl: ""
```

jbygdell commented on charts/htsget-rs/values.yaml:
```diff
@@ -0,0 +1,164 @@
+  forwardHeaders: true
+
+  # Header and file url
+  index: "http://host.minikube.internal:8443/s3/"
```
```suggestion
  index: ""
```

jbygdell commented on charts/htsget-rs/values.yaml:
```diff
@@ -0,0 +1,164 @@
+  # Header and file url
+  index: "http://host.minikube.internal:8443/s3/"
+  # file: "http://host.minikube.internal:8443/s3-encrypted/"
+  file: "http://host.minikube.internal:8443/s3/"
```
```suggestion
  file: ""
```

jbygdell commented on .github/integration/scripts/values.yaml:
```diff
@@ -0,0 +1,164 @@
+
+c4gh:
+    # The secret name that contains the c4gh keys
+    secretName: "c4gh"
```
```suggestion
    secretName: ""
```

jbygdell commented on charts/htsget-rs/templates/configmap.yaml:
```diff
@@ -0,0 +1,49 @@
+kind: ConfigMap
+metadata:
+  name: {{ template "htsget-rs.fullname" . }}-config
+data:
```
Service info config s are missing, [example](https://github.com/umccr/htsget-rs/tree/main/htsget-config#service-info-config)

jbygdell commented on charts/htsget-rs/templates/configmap.yaml:
```diff
@@ -0,0 +1,49 @@
+{{- else }}
+    ticket_server_addr = "127.0.0.1:8080"
+    ticket_server_cors_allow_origins = "All"
+
```
```suggestion
# disable the internal data server
data_server_enabled = false

```

jbygdell commented on charts/htsget-rs/templates/deployment.yaml:
```diff
@@ -0,0 +1,138 @@
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "htsget-rs.serviceAccountName" . }}
```
```suggestion
```

aaperis replied:
Kept for this version.

jbygdell commented on README.md:
````diff
@@ -1,2 +1,113 @@
+```sh
+curl -k https://htsget-htsget-rs:8080/reads/service-info
+```
+to get the htsget response.
````
```suggestion
```

This needs to be replaced with detailed information about the values file and what parts are required and used where.
[like this](https://github.com/grafana/helm-charts/tree/main/charts/grafana#readme)

aaperis replied:
Indeed, now done. I kept the original README contents but placed them in a different file in case someone finds them useful.

jbygdell commented on charts/htsget-rs/values.yaml:
```diff
@@ -0,0 +1,164 @@
+
+# For development purposes it is also possible to pass htsget-rs specific configuration by pasting the
+# contents of the toml config file (https://github.com/umccr/htsget-rs/tree/crypt4gh/htsget-config) below.
+# Use nident 2. (experimental, will not work with c4gh keys and mtls certificates)
```
```suggestion
```

aaperis replied:
Tried to rephrase it a bit so that it is clearer if someone wants to deploy htsget-rs for a use case outside gdi.

jbygdell commented on charts/htsget-rs/values.yaml:
```diff
@@ -0,0 +1,164 @@
+#   forward_headers = true
+#   [resolvers.storage.endpoints]
+#   index = ""
+#   file = ""
```
Instead of doing the if/else in the `configmap.yaml` put a default working example here.
Remember that you are creating a helm chart for a public project that also should work outside of GDI

For GDI a specific readme should be placed in the straterkit-htsget repo.

aaperis replied:
I have now removed any configMap logic and made the app configurable only via a toml syntax directly in the values yaml file. Any additional resources needed for passing certificates or c4gh keys are controlled by boolean variables and the filepaths that I have added as variables. This seems cleaner and more generic. Hopefully the charts can be usable for purposes outside of GDI in this way!

aaperis committed:
add manual release for chart

jbygdell commented on .github/integration/scripts/values.yaml:
```diff
@@ -0,0 +1,164 @@
+  enabled: true
+  className: "nginx"
+  annotations: {}
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
```
```suggestion
    # cert-manager.io/cluster-issuer: 
```

Better to replace these with something that is not obsolete if you want some examples.

jbygdell commented on .github/integration/scripts/values.yaml:
```diff
@@ -0,0 +1,164 @@
+
+ingress:
+  enabled: true
+  className: "nginx"
```
```suggestion
  className: ""
```

jbygdell commented on .github/integration/scripts/values.yaml:
```diff
@@ -0,0 +1,164 @@
+    - host: htsget.local
+      paths:
+        - path: /
+          pathType: ImplementationSpecific
```
```suggestion
          pathType: Prefix
```
Prefix would be more appropriate since we want to sen all queries to the backend

jbygdell commented on .github/integration/scripts/values.yaml:
```diff
@@ -0,0 +1,164 @@
+  # runAsNonRoot: true
+  # # runAsUser: 1000
+  # seccompProfile:
+  #           type: "RuntimeDefault"
```
```suggestion
containerSecurityContext:
  allowPrivilegeEscalation: false
  capabilities:
    drop:
    - ALL
  privileged: false

podSecurityContext:
  fsGroup: 1000
  runAsNonRoot: true
  runAsUser: 1000
  seccompProfile:
    type: "RuntimeDefault"
```

Better to have sane working defaults

aaperis committed:
debug tests

aaperis committed:
create certs upon issuer definition

aaperis committed:
use image with hardened security

- update default image
- enforce security context

aaperis committed:
enable security context in test

aaperis committed:
refine security policy

aaperis committed:
remove ingress.tlsEnabled variable

- revert to enabling tls with non-empty matrix
- fix identation for extra annotation in ingress yaml
- update readme instructions

aaperis committed:
refine certificate

- client auth not needed
- strengthen encryption algorithm

aaperis committed:
update chart details

- update description
- add sources
- bump appVersion

aaperis committed:
fix typo

aaperis committed:
cleanup values yaml

aaperis committed:
refine deployment yaml

- fix typo
- set path variables as required

jbygdell commented on charts/htsget-rs/templates/ingress.yaml:
```diff
@@ -0,0 +1,70 @@
+  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
```
There is absolutely no reason to support anything earlier than `1.19`.
If you want this chart to be usable as is on any relevant version of k8s, ensure that you have correct separation between (>=1.19  <=1.24) and >=1.25.

aaperis replied:
This should be done now.

aaperis committed:
restructure how htsget is configured

- use toml config inplace in values yaml
- configure file mount details from values yaml
- update integration tests accordingly

aaperis committed:
fix integration tests

- remove certificate filepaths if tls is disabled

aaperis committed:
restrict k8s version to >=1.19

aaperis committed:
add legacy PSP support

aaperis committed:
update README

- add config description
- default tls to disabled

aaperis committed:
add documentation for local test environment

aaperis committed:
simplify config filename

jbygdell commented on charts/htsget-rs/templates/deployment.yaml:
```diff
@@ -0,0 +1,143 @@
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- if .Values.rbacEnabled}}
```
```suggestion
      {{- if and .Values.rbacEnabled semverCompare "<v1.25.0" .Capabilities.KubeVersion.Version}}
```
Service account is only needed for the old versions.

jbygdell commented on howtotest.md:
```diff
@@ -0,0 +1,124 @@
+
+### Testing with sda-download
+
+Start by checking out [this branch}(https://github.com/GenomicDataInfrastructure/starter-kit-storage-and-interfaces/tree/featute/test-htsget-reencrypt) of the GDI starter-kit-storage&interfaces repository. This branch contains the re-encryption of file headers in the sda-download service.
```
This should not point to a specific branch.

aaperis replied:
File removed.

jbygdell commented on .github/integration/scripts/deploy_charts.sh:
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -e
+
+if [ "$1" == "true" ]; then
+    echo "TLS enabled"
+    sed -i 's/# .*\(ticket_server_tls.key.*\)/\1/' .github/integration/scripts/values.yaml
+    sed -i 's/# .*\(ticket_server_tls.cert.*\)/\1/' .github/integration/scripts/values.yaml
+else
+    echo "TLS disabled"
+    sed -i 's/\(ticket_server_tls.key.*\)/# \1/' .github/integration/scripts/values.yaml
+    sed -i 's/\(ticket_server_tls.cert.*\)/# \1/' .github/integration/scripts/values.yaml
+fi
```
```suggestion
if [ "$1" == "false" ]; then
    echo "TLS disabled"
    sed -i '/ticket_server_tls.*/d' .github/integration/scripts/values.yaml
fi
```

TLS is the default value so you only need to remove it if it is disabled.

aaperis replied:
This is handy when running local tests for development, I'd prefer to keep it here.

jbygdell commented on .github/integration/scripts/values.yaml:
```diff
@@ -0,0 +1,183 @@
+
+  # Path where the sensitive files can be found.
+  # These should match the path in the htsget config below.
+  c4ghPath: ""
```
```suggestion
  c4ghPath: "/c4gh"
```

jbygdell commented on .github/integration/scripts/values.yaml:
```diff
@@ -0,0 +1,183 @@
+  ticket_server_tls.cert = "/tls/tls.crt"
+
+  # disable the internal data server
+  data_server_enabled = false
```
```suggestion
  data_server_enabled = true
```

jbygdell commented on .github/integration/scripts/values.yaml:
```diff
@@ -0,0 +1,183 @@
+  # of creating one on-the-fly when handling encrypted data
+  predefined: false
+  # The name for the secret containing c4gh keys.
+  secretName: ""
```
```suggestion
  secretName: "c4gh"
```

jbygdell commented on .github/integration/scripts/values.yaml:
```diff
@@ -0,0 +1,183 @@
+c4gh:
+  # Set to true if htsget will use a predefined c4gh key-pair instead
+  # of creating one on-the-fly when handling encrypted data
+  predefined: false
```
```suggestion
  predefined: true
```

jbygdell commented on .github/integration/scripts/dependencies.sh:
```diff
@@ -0,0 +1,42 @@
+.c4gh.secretName = "c4gh" |
+.htsget.c4ghPath = "/c4gh" |
+.htsget.tlsPath = "/tls"
+' .github/integration/scripts/values.yaml
```
```suggestion
```

All of these can be set directly in the values file instead.

jbygdell commented on charts/htsget-rs/values.yaml:
```diff
@@ -0,0 +1,183 @@
+  # ticket_server_tls.cert = "/tls/tls.crt"
+
+  # disable the internal data server
+  data_server_enabled = false
```
We should have a working example config with the parts needed when the dataserver is enabled

aaperis replied:
Done. In this first iteration I had not considered to cover all configuration setups but yes, it makes sense to have the most generic usecase as the main example here instead of the GDI one. I now added logic for enabling local storage for the data server and for separating the data server to a different service and ingress.

jbygdell reviewed (dismissed):
The `howtotest.md` isn't really useful for the end user.
If this chart had been located in the GDI organization then maybe it would have been warranted, but probably not.

I suggest moving it to a wiki page instead.

aaperis committed:
simplify integration tests

aaperis committed:
service account only needed for old k8s versions

aaperis committed:
add dataserver local storage config support

aaperis committed:
update readme

aaperis committed:
remove local testing documentation

aaperis committed:
update test configuration

aaperis committed:
no need for localPath if dataserver is disabled

aaperis committed:
simplify values.yaml

jbygdell commented on charts/htsget-rs/values.yaml:
```diff
@@ -0,0 +1,180 @@
+
+  [[resolvers]]
+  regex = ".*"
+  substitution_string = "$0"
```
```suggestion
  substitution_string = "$0"
  
  [resolvers.object_type]
  # if set to true, responseUrl must point to an endpoint serving encrypted data
  send_encrypted_to_client = true
  # if c4gh keys are not set, they will be generated on the fly
  # only custom keys without a passphrase are supported
  private_key = "/c4gh/c4gh.sec.pem"
  public_key = "/c4gh/c4gh.pub.pem"

  [resolvers.storage]
  # The url that will be used for the client's url
  response_url = "http://example.com"
  forward_headers = true
  # if set to true, the client will not check the certificate validity
  danger_accept_invalid_certs = true

  [resolvers.storage.endpoints]
  index = "http://example.com"
  file = "http://example.com"
```


Main values file is missing S3 storage example

aaperis replied:
The dataserver is incompatible with the above resolvers configuration that is why they are omitted. Further examples for these configs exist under `.github/integration/scripts` and referenced in the README. The same goes for s3 storage.

aaperis committed:
add dataserver svc and ingress

aaperis committed:
cleanup NOTES to users

aaperis committed:
update README

aaperis committed:
add logic local dataserver tests

aaperis committed:
add local dataserver tests

aaperis committed:
break out manifests to files

aaperis committed:
don't print dataserver info if not enabled

aaperis committed:
add basic support for s3

aaperis committed:
add test dependencies for s3

aaperis committed:
add tests for s3 storage

aaperis committed:
update readme

aaperis committed:
minimize test cases run for s3

aaperis committed:
fix tests

aaperis committed:
add integration tests

- for dataserver
- for s3storage

MalinAhlberg commented on .github/integration/scripts/data-server-config.toml:
```diff
@@ -0,0 +1,29 @@
+
+# # [[resolvers]]
+# # regex = "(.*)"
+# # substitution_string = "$1"
```
Remove the commented lines?

aaperis replied:
Good catch!

MalinAhlberg commented on .github/integration/test-data/bam/htsnexus_test_NA12878.bam.blocks.yaml:
```diff
@@ -0,0 +1,95 @@
+---
```
Maybe add a readme or similar in this folder, stating where the data comes from?

aaperis replied:
Good point @MalinAhlberg, now done.

MalinAhlberg reviewed (dismissed):
Very good work, @aaperis!

Also thanks for the overview on zoom :pray: . I think it looks good, just left two minor suggestions. They're not stopping my approval, though.

aaperis committed:
apply suggestions from review

MalinAhlberg reviewed (approved):
:ok_hand:

pahatz reviewed (approved):
Works great!
